internal/provider/resource_cloudflare_device_posture_rule_test.go

```diff
@@ -216,6 +216,43 @@ func TestAccCloudflareDevicePostureRule_DiskEncryption(t *testing.T) {
 	})
 }
 
+func TestAccCloudflareDevicePostureRule_CrowdstrikeS2S(t *testing.T) {
+	// Temporarily unset CLOUDFLARE_API_TOKEN if it is set as the Access
+	// service does not yet support the API tokens and it results in
+	// misleading state error messages.
+	if os.Getenv("CLOUDFLARE_API_TOKEN") != "" {
+		t.Setenv("CLOUDFLARE_API_TOKEN", "")
+	}
+
+	rnd := generateRandomResourceName()
+	name := fmt.Sprintf("cloudflare_device_posture_rule.%s", rnd)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			testAccPreCheck(t)
+		},
+		ProviderFactories: providerFactories,
+		CheckDestroy:      testAccCheckCloudflareDevicePostureRuleDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCloudflareDevicePostureRuleConfigDiskEncryption(rnd, accountID),
```

Comment:
this needs to be a new method, like `testAccCloudflareDevicePostureRuleConfigCrowdstrikeS2S` to get the correct config.

```
func testAccCloudflareDevicePostureRuleConfigCrowdstrikeS2S(rnd, accountID string) string {
	return fmt.Sprintf(`
resource "cloudflare_device_posture_rule" "%[1]s" {
	account_id                = "%[2]s"
	name                      = "%[1]s"
	type                      = "crowdstrike_s2s"
	description               = "My description"
	schedule                  = "24h"
	expiration                = "24h"

	match {
		platform = "mac"
	}

	input {
		connection_id = "af8d87a7-1272-4932-92b8-208ffbead88e"
		version_operator = "<>"
		version = true
	}
}
`, rnd, accountID)
}
```